sycl/test/basic_tests/select_device.cpp

```diff
@@ -0,0 +1,82 @@
+// RUN: %clangxx -fsycl -fsycl-targets=%sycl_triple %s -o %t.out
+// RUN: env SYCL_BE=%sycl_be %t.out
+// RUN: env SYCL_DEVICE_TYPE=HOST %t.out
+// RUN: env SYCL_DEVICE_TYPE=CPU %t.out
+// RUN: env SYCL_DEVICE_TYPE=GPU %t.out
+// RUN: env SYCL_DEVICE_TYPE=ACC %t.out
+// RUN: env SYCL_DEVICE_TYPE=GPU SYCL_BE=%sycl_be %t.out
+// RUN: env SYCL_DEVICE_TYPE=CPU SYCL_BE=%sycl_be %t.out
+// RUN: env SYCL_DEVICE_TYPE=HOST SYCL_BE=%sycl_be %t.out
+//
+// Checks that only designated plugins are loaded when SYCL_FORCE_PI is set.
+// Checks that all different device types can be acquired from select_device()
+// regardless of env var setting SYCL_BE and/or SYCL_DEVICE_TYPE.
+
+#include <CL/sycl.hpp>
+#include <iostream>
+
+using namespace cl::sycl;
+
+int main() {
+  const char *pis = std::getenv("SYCL_FORCE_PI");
+  const char *be = std::getenv("SYCL_BE");
+  std::string forcedPIs;
+  std::string forcedBE;
+  if (pis) {
+    forcedPIs = pis;
+  }
+  if (be) {
+    forcedBE = be;
+  }
```

Comment:
Could you please clarify why this is done?

Reply:
const char* be is reading the env var SYCL_BE.
So, this code sets the string forcedBE only when this env var is set.
Otherwise (the be pointer is null), the string remains empty.